cmake-modules/AzureDoxygen.cmake

```diff
@@ -19,7 +19,15 @@ function(generate_documentation PROJECT_NAME PROJECT_VERSION)
         set(DOXYGEN_RECURSIVE YES)
         set(DOXYGEN_USE_MDFILE_AS_MAINPAGE ./README.md)
         set(DOXYGEN_INLINE_SOURCES YES)
+        # Skip generating docs for json, test and private files.
+        set(DOXYGEN_EXCLUDE_PATTERNS
+            json.hpp
+            test
+            *private*)
+        # Skip documenting Details symbols (all from ::Details namespace)
+        set(DOXYGEN_EXCLUDE_SYMBOLS Details)
         set(DOXYGEN_IGNORE_PREFIX az_ AZ_)
+        set(DOXYGEN_)
```

Comment:
I am curious, what does this do?

Reply:
This is something that shouldn't be here!  hahaha.. I will create PR to remove it.
Thanks!